File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        temp = ListNode()
        pointer = temp
        carry = 0
        while l1!=None or l2!=None or carry!=0:
            val1=l1.val if l1!=None else 0
            val2=l2.val if l2!=None else 0
            sum = val1 + val2 + carry
            digit = sum%10
            carry = sum//10
            pointer.next=ListNode(digit)
            pointer = pointer.next
            l1=l1.next if l1!=None else None
            l2=l2.next if l2!=None else None
        return temp.next
    
        

File: 0002-add-two-numbers/0002-add-two-numbers.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* dummy = new ListNode(-1);
        int carry = 0;
        ListNode* curr = dummy;
        while (l1 != nullptr || l2 != nullptr) {
            int sum = carry;
            if (l1)
                sum += l1->val;
            if (l2)
                sum += l2->val;
            ListNode* newnode = new ListNode(sum % 10);
            carry = sum / 10;
            if (l1)
                l1 = l1->next;
            if (l2)
                l2 = l2->next;
            curr->next = newnode;
            curr = curr->next;
        }
        if (carry) {
            curr->next = new ListNode(carry);
        }
        return dummy->next;
    }
};